plugins/experimental/maxmind_acl/maxmind_acl.cc

```diff
@@ -0,0 +1,117 @@
+/*
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+#include "mmdb.h"
+
+static int
+config_handler(TSCont cont, TSEvent event, void *edata)
+{
+  TSMutex mutex;
+
+  mutex = TSContMutexGet(cont);
+  TSMutexLock(mutex);
```

Comment:
Hmm won't the core lock the continuation mutex before calling the continuation function?

Reply:
Does it? I was just going by what had been done previously, and I have been able to reload the config fine with this code, so it appears to work

Reply:
Oh Id wager that it doesnt do it for this case because this a Mgmt cont. In the normal case, no I dont do the lock/unlock, but this is getting hit from a config reload so we can reload the plugin configuration so management probably doesnt lock that mutex, but we need to lock it to do the reload

Reply:
Hmm, weirdly, it's implementation-dependent whether pthread mutexes are recursive by default ( https://linux.die.net/man/3/pthread_mutexattr_settype ).  Since we never set the mutex type that I can see, our mutexes may or may not be recursive.

Reply:
It looks like the mutex does get locked before the continuation func is called:  https://github.com/apache/trafficserver/blob/da888769f4a83c731e08fdddf5d038cf64fa8b06/proxy/InkAPIInternal.h#L304 .  I guess we just better hope nobody builds TS with a pthreads implementation where mutexes are not recursive by default.

Reply:
yea, we'd hit this in a couple of plugins.

Meanwhile while poking at this I did run in to some other issue that appears to be related to yaml usage, but only while doing a reload after the remap has changed.  However removing the mutex usage by itself appears to be fine

Reply:
Looking at other plugins, it appears this is only supported when they are running as global plugins anyway.  So for now I will go back when I get a chance and rip out all of the config reloading stuff since it just flat out doesnt work for remap plugins the way that mgmt functionality is implemented anyway